### PR TITLE
UPP store tests

### PR DESCRIPTION
**Pull request by srallen**
Package: lib-classifier

Describe your changes:
Adds tests for the UPP store in the classifier which I noticed were missing while I was working on it this week.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

File: packages/lib-classifier/src/store/UserProjectPreferencesStore.spec.js [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+  it('should set project preferences if there is a user and a project', function (done) {
+    rootStore = RootStore.create({
+      projects: ProjectStore.create(),
+      userProjectPreferences: UserProjectPreferencesStore.create()
+    }, { authClient: authClientStubWithUser, client: clientStub })
+    rootStore.projects.setActive(project.id)
+      .then(() => {
+        const uppInStore = rootStore.userProjectPreferences.active
+        expect(uppInStore.toJSON()).to.deep.equal(upp)
+      }).then(done())
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/UserProjectPreferencesStore.spec.js]: This passes the return value of `done()` to `then` as a callback function. It should be `.then(done, done)` to catch failing tests.

File: packages/lib-classifier/test/stubPanoptesJs/stubPanoptesJs.js [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
   return {
     panoptes: {
-      get: function (url, query) {
-        // Assuming url is in the format '/resource_type/resource_id'
-        const endpoint = url.split('/')[1]
-        const response = {}
-        response[endpoint] = [factories[endpoint]]
-        return Promise.resolve({ body: response })
-      }
+      get: (url) => createResponse(url, factories),
```
Comment (eatyourgreens) [inline packages/lib-classifier/test/stubPanoptesJs/stubPanoptesJs.js]: Would it be worth wrapping these in spies, for convenience?

**Comment by eatyourgreens** [comment]
I think you might have accidentally written evergreen tests for promises by passing `done()` to `then`. You could use `.then(done, done)` to catch failing tests, or use `await`. I find myself referring to [this post about testing promises with mocha](https://wietse.loves.engineering/testing-promises-with-mocha-90df8b7d2e35) whenever I'm testing API calls.

File: packages/lib-classifier/test/stubPanoptesJs/stubPanoptesJs.js (hunk 2 above)
Comment (srallen) [reply packages/lib-classifier/test/stubPanoptesJs/stubPanoptesJs.js]: Probably? I'd like to rethink what this helper function is doing separately though and build it out for mocking the other HTTP methods. And perhaps abstract it out to be a part of panoptes.js lib instead and available to other consumers.

File: packages/lib-classifier/src/store/UserProjectPreferencesStore.spec.js (hunk 1 above)
Comment (srallen) [reply packages/lib-classifier/src/store/UserProjectPreferencesStore.spec.js]: Thanks for catching that. I usually return the promise, but using the done callback just made more sense here.

**Comment by srallen** [comment]
@eatyourgreens this is ready for another review.

**Pull request merged**